benchmarl/environments/magent/common.py

```diff
@@ -0,0 +1,125 @@
+#  Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+#  This source code is licensed under the license found in the
+#  LICENSE file in the root directory of this source tree.
+#
+
+from typing import Callable, Dict, List, Optional
+
+from torchrl.data import CompositeSpec
+from torchrl.envs import EnvBase, PettingZooWrapper
+
+from magent2.environments import (
+    adversarial_pursuit_v4,
+    battle_v4,
+    battlefield_v5,
+    combined_arms_v6,
+    gather_v5,
+    tiger_deer_v4
+)
+
+from benchmarl.environments.common import Task
+
+from benchmarl.utils import DEVICE_TYPING
+
+
+class MAgentTask(Task):
+    """Enum for MAgent2 tasks."""
+
+    ADVERSARIAL_PURSUIT = None
+    BATTLE = None
+    BATTLEFIELD = None
+    COMBINED_ARMS = None
+    GATHER = None
+    TIGER_DEER = None
+
+    def get_env_fun(
+            self,
+            num_envs: int,
+            continuous_actions: bool,
+            seed: Optional[int],
+            device: DEVICE_TYPING,
+    ) -> Callable[[], EnvBase]:
+
+        return lambda: PettingZooWrapper(
+            env=self.__get_env(),
+            return_state=True,
+            seed=seed,
+            done_on_any=True,
```

Comment:
Setting `done on any` means that the global task done when any agent in any group is done.
From the associated issue I understood that MAgent2 can have done agents, so maybe we need `done on any=False`?

Reply:
You're right. I fixed it.